modules/imgproc/src/matchcontours.cpp

```diff
@@ -62,6 +63,11 @@ double cv::matchShapes(InputArray contour1, InputArray contour2, int method, dou
             double ama = fabs( ma[i] );
             double amb = fabs( mb[i] );
 
+            if (ama > 0)
+                anyA = true;
```

Comment:
No, that's not the same. That can set anyA from true to false, which we don't want.

`anyA |= ama >0` would work, but then you have to depend on people remember what |= does, and noticing the |, which a quick skim would probably miss.